nixos/modules/system/boot/loader/grub/install-grub.pl

```diff
@@ -370,6 +373,28 @@ sub GrubFs {
             fi
         ";
     }
+
+    rmtree("$bootPath/theme") or die "cannot clean up theme folder in $bootPath\n" if -e "$bootPath/theme";
+
+    if ($theme) {
+        # Copy theme
+        rcopy($theme, "$bootPath/theme") or die "cannot copy $theme to $bootPath\n";
+        $conf .= "
+            # Sets theme.
+            set theme=" . ($grubBoot->path eq "/" ? "" : $grubBoot->path) . "/theme/theme.txt
```

Comment:
Is it an issue with grub, though ugly, if the path starts with two slashes?

If it isn't we should drop the unnecessary ternary, so the code is easier to understand.

Alternatively, there may be something perl-y that handles paths concatenation for us, rather than doing the hard task ourselves?

Reply:
I copied this from the other code, just following the pattern here... not my department :P

https://github.com/NixOS/nixpkgs/pull/83678/files#diff-3f0e9813c2baa6a4e7f8dbe5fd38ea56L364

Reply:
Ah, then don't worry too much, even though the question still stands in a more global manner.